2019/04/fourth_day.rb

```diff
@@ -1,11 +1,30 @@
-input = 353096..843212
-
-# find out, how many strings from 353096-843212 are potential passwords
-# Criteria:
-# 6 digits (all of them)
-# two adjacent digits are the same
-# digits never decrease, but increase or stay the sam
-#
-# plan:
-# iterate over all possible input-strings
-# check for criteria. if one fails: next string
+
+def is_number_potential_password?(number)
+  # has six digits
+  has_six_digits = number.to_s.length == 6
+  # two adjacent digits are the same
+  two_adjacent_digits = false
+  number_string = number.to_s
+  if number_string[0] == number_string[1] || number_string[1] == number_string[2] || number_string[2] == number_string[3] || number_string[3] == number_string[4] || number_string[4] == number_string[5]
```

Comment:
Checkout the `each_cons` method!

Sth. like:

```ruby
two_adjacent_digits = number_string.chars.each_cons(2).any? { |a,b| a == b }
```

Reply:
Awesome, thank you!